velox/exec/SortBuffer.cpp

```diff
@@ -285,6 +285,10 @@ void SortBuffer::spillOutput() {
     // Already spilled.
     return;
   }
+  if (numOutputRows_ == sortedRows_.size()) {
+    // All the output has been produced.
+    return;
```

Comment:
I should add this check here in the order by output spill PR, thanks for the fix.
Curiously, the `spiller_->spill(spillRows)` returns directly if the `spillRows` vector is empty when `numOutputRows_ == sortedRows_.size()`. Is the `concurrent test` failed by the following operations, `data_->clear() & finishSpill()`?
https://github.com/facebookincubator/velox/blob/7a9c4b957a713059862324821b5e7a7cbb5f9657/velox/exec/Spiller.cpp#L494-L500

Reply:
It fails with a check expect the spilled partition is not empty.

Reply:
Should we change the `if (rows.emepty()) return;` to `VELOX_CHECK(!rows.empty())` in `Spiller::spill(std::vector<char*>& rows)`?

Reply:
@duanmeng yes, we could. I will leave this after a spiller constructor refactor. thanks!

Reply:
@duanmeng  I added the check. Thanks!